che-incubator/che-code
# Devfile Commands support

Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

The PR adds the Che-Commands extension. It detects the Commands defined in a Devfile and represents them as VS Code Tasks. The environment variables substitution is supported for the `exec.workingDir` property.

The PR closes https://github.com/eclipse/che/issues/21030

#### How to test
1. Spin up a DevWorkspace using the following link:
https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#https://github.com/azatsarynnyy/java-spring-petclinic/tree/devfilev2
2. In the command palette type a `task ` shortcut and choose a `che` task type.
3. Run `prepare-database` command. It's running in the `mariadb` component. See the output.
4. Run `maven-build` command. It's running in the `tools` component. See the output.

![che-code-task](https://user-images.githubusercontent.com/1636395/171450174-4ccff84d-c048-48a1-9b56-a82ea854a1dd.gif)
